### PR TITLE
Just polished #83 (Add OpenSSL-Compatible PKCSv5#1)

### PR DESCRIPTION
**Pull request by karel-m**
#83 + `static` fix + tabs converted to spaces

### CODE REVIEW

**Comment by sjaeckel** [comment]
There are still tests missing

**Comment by karel-m** [comment]
I have added some alg1 + alg1_openssl tests to `pkcs_5_test.c`. However those are not official test vectors, just the values that it gave me on my laptop.

Anyway, tests are there, once anybody have better test vectors they can be easily added.

**Comment by karel-m** [comment]
Hmm, travis:
```
building...testing...*** stack smashing detected ***: ./test terminated
```

**Comment by karel-m** [comment]
@sjaeckel after rebase on the latest develop travis-ci is all green, do you agree with merging this one?

File: demos/openssl-enc.c [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+   unsigned char keyiv[KEY_LENGTH + IV_LENGTH];
+   unsigned long keyivlen = (KEY_LENGTH + IV_LENGTH);
+   unsigned char *key, *iv;
+
+   /* Check proper number of cmdline args */
+   if(argc < 5 || argc > 6)
+      BARF("Invalid number of arguments");
+
+   /* Check proper mode of operation */
+   if     (!strncmp(argv[1], "enc", sizeof("enc")))
```
Comment (sjaeckel) [inline demos/openssl-enc.c]: erm, \*cough\* `sizeof("enc")`

File: demos/openssl-enc.c [hunk 2]
```diff
@@ -0,0 +1,381 @@
+/*
+ * Demo to do the rough equivalent of:
+ *
+ *    openssl enc -aes-256-cbc -pass pass:foobar -in infile -out outfile -p
+ *
+ * Compilation:
+ *
+ *    $(CC) -I /path/to/headers -L .../libs \
```
Comment (sjaeckel) [inline demos/openssl-enc.c]: I think it'd make sense to have a make target that builds this...

File: demos/openssl-enc.c [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      /* We are PADDING this block (and therefore adding bytes) */
+      /* The pad value in PKCS#7 is the number of bytes remaining in
+         the block, so for a 16-byte block and 3 bytes left, it's
+         0x030303.  In the oddball case where nb is an exact multiple
+         multiple of block_length, set the padval to blocksize (i.e.
+         add one full block) */
+      padval = (unsigned char) (block_length - (nb % block_length));
+      padval = padval ? padval : block_length;
+
+      XMEMSET(buf->pad+nb, padval, padval);
```
Comment (sjaeckel) [inline demos/openssl-enc.c]: either use `XMEMSET` etc.

File: demos/openssl-enc.c [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+int parse_openssl_header(FILE *in, unsigned char *out)
+{
+   unsigned char tmp[SALT_LENGTH];
+   if(fread(tmp, 1, sizeof(tmp), in) != sizeof(tmp))
+      return CRYPT_ERROR;
+   if(memcmp(tmp, salt_header, sizeof(tmp)))
+      return CRYPT_ERROR;
+   if(fread(tmp, 1, sizeof(tmp), in) != sizeof(tmp))
+      return CRYPT_ERROR;
+   memcpy(out, tmp, sizeof(tmp));
```
Comment (sjaeckel) [inline demos/openssl-enc.c]: or use the original standard functions, but please don't mix...

**Pull request merged**